### DISCUSSION TITLE
Gradient Image (Transform command from IM 7 to Magick.Net)

### DISCUSSION THREAD

**Discussion by jaka1-newbie** [dlemstra/Magick.NET#1008] [Help]
Please help to transform command.
It do gradient at image from 1 color to another.
I can't find any other solutions, only this from IM7 

`back.png -set option:dims "%wx%h" ( -size "%[dims]" gradient:"{-Variable.backColor-}-{-Variable.textColor-}" ) ( -clone 0 -alpha extract ) -delete 0 -alpha off -compose copy_opacity -composite back.png`

**Comment by dlemstra** [comment]
What have to tried so far? Most options are methods on the `MagickImage` or the `MagickImageCollection` class.

**Reply by jaka1-newbie** [reply]
I don't understand how to set options and didn't find gradient command

**Reply by dlemstra** [reply]
You can set an option with `image.Settings.SetDefine(name, value)`. And if you want to read a gradient you will want to do this: `new MagickImage("gradient:color1-color2")`.

**Comment by jaka1-newbie** [comment]
Can Magick.Net do gradient border ?

**Reply by dlemstra** [reply]
This library can do everything ImageMagick does. Not sure what yo are asking.

**Comment by jaka1-newbie** [comment]
Something like this 
![35c9711bc038263850b2b7b221066ff1 1](https://user-images.githubusercontent.com/67605008/130214672-3781d887-9c72-44b9-b7cf-636221efcb6d.jpg)

**Reply by dlemstra** [reply]
A possible solution would be this:

```C#
using (var image = new MagickImage("input.png"))
{
    var size = 10;
    using (var gradient = new MagickImage("gradient:green-pink", image.Width + size, image.Height + size))
    {
        gradient.Composite(image, Gravity.Center);
        gradient.Write("output.png");
    }
}
```

**Reply by jaka1-newbie** [reply]
As i understood we create new MagickImage.
So can we create gradient for 1 letter? 
Like this 
https://prnt.sc/1rbcsiu
This is monogram font.

**Reply by dlemstra** [reply]
I have no idea what you are trying to ask me.

**Reply by jaka1-newbie** [reply]
I want to create Image with 1 Letter 
(This is letter from monogram font) and want color of this letter - gradient.
https://prnt.sc/1rbcsiu

**Comment by jaka1-newbie** [comment]
From https://prnt.sc/1rbipek
To https://prnt.sc/1rbiqzb

**Comment by jaka1-newbie** [comment]
convert -size 320x100 xc:lightblue -font Candice -pointsize 72 -tile gradient:   -annotate +28+68 'Anthony'   font_gradient.jpg

**Reply by dlemstra** [reply]
That command would translate into this:

```C#
using (var image = new MagickImage("xc:lightblue", 320, 100))
{
    using (var gradient = new MagickImage("gradient:", 320, 100))
    {
        image.Settings.Font = "Candice";
        image.Settings.FontPointsize = 72;
        image.Settings.FillPattern = gradient;

        // You need Gravity.Undefined for now due to a bug, you won't need it in the next release.
        image.Annotate("Anthony", new MagickGeometry("+28+68"), Gravity.Undefined);

        image.Write("font_gradient.jpg");
    }
}
```

**Reply by jaka1-newbie** [reply]
Yes! Thx! 
How to make more difficult gradients and change direction?

**Reply by dlemstra** [reply]
What do you mean by a more difficult gradient?
